### ISSUE TITLE
Define the behavior of validation during training

### ISSUE THREAD

**Issue by wangkuiyi** [sql-machine-learning/sqlflow#675]
In https://github.com/sql-machine-learning/sqlflow/issues/660#issue-480569126,

@weiguoz proposed two solutions:

1. Add the `VALIDATE ON` clause to `TRAIN`:

   ```sql
   SELECT ... TRAIN DNNClassifier VALIDATE ON (SELECT ...)
   ```

   However, it is error-prone that the training dataset from the first `SELECT ...` and the validation dataset from the second `SELECT ...` have different data schema.

1. Keep only the first `SELECT ...` in the above example, and split the returned rows into the training and the validation datasets.

   However, as reminded by @sperlingxx, it is often that we cannot split by rows randomly. For example, we might need to split data by user IDs.

In short, we don't have a clear idea of how to define the behavior of validation yet.

**Comment by typhoonzero** [comment]
Already implemented, see details at: https://github.com/sql-machine-learning/sqlflow/issues/1111
